### DISCUSSION TITLE
Making KorGE video tutorials the easy way in macOS

### DISCUSSION THREAD

**Discussion by soywiz** [korlibs/korge#943] [Tutorials]
## Setting up the interactive mode

1. First, we have to run KorGE in interactive mode, by running the `runJvmAutoreload` gradle task.
2. Then we have to maximize the korge window to the right or the left, and then put the Intellij IDE on the opposite side. To be consistent we suggest to use the left side for the editor, and the right side for the preview.
3. For each change that you want to be hot-updated, press `cmd+s` to save the file. Once the file is saved, the plugin will detect a change and will recompile and hot reload the visible scene.

https://user-images.githubusercontent.com/570848/186721884-64abbbc7-76e9-4976-96e1-18609d259e1e.mp4

## Recording the screen

1. We can then start recording the screen. For that, we can use the integrated screen recorder by pressing `cmd`+`shift`+`5`.

## Uploading to github

1. You can create a new Tutorial in the tutorials section here: https://github.com/korlibs/korge/discussions/new?category=tutorials
2. Then you can drag&drop your video file to the markdown text area. The file must be 10 MB or less, so if your video is bigger, consider reducing it with [handbrake](https://handbrake.fr/), or any other software.

**Comment by JDSCaram** [comment]
Hey, @soywiz My auto reload does not work, Look 👇 
Do I have to do anything else?

https://user-images.githubusercontent.com/18143517/186941872-c4a51684-c363-4548-a9d7-110746001e46.mp4

**Reply by soywiz** [reply]
Yes, sorry. You have to press `cmd+s` to save the file on each change :)

Updated the description too as point `3.`
